week-4/E-arrays-of-objects/exercise-2.js

```diff
@@ -7,50 +7,72 @@ Each destination has a name, a distance from Glasgow, and a list of transportati
 3) Print in the console all the destination names more than 300 kms far away and reachable by train.
 */
 
-
 var destination1 = {
-    destinationName: "Edinburgh",
-    distanceKms: 80,
-    transportations: ["car", "bus", "train"]
+  destinationName: "Edinburgh",
+  distanceKms: 80,
+  transportations: ["car", "bus", "train"]
 };
 
 var destination2 = {
-    destinationName: "London",
-    distanceKms: 650,
-    transportations: ["car", "bus", "train"]
+  destinationName: "London",
+  distanceKms: 650,
+  transportations: ["car", "bus", "train"]
 };
 
 var destination3 = {
-    destinationName: "Paris",
-    distanceKms: 900,
-    transportations: ["train", "plane"]
+  destinationName: "Paris",
+  distanceKms: 900,
+  transportations: ["train", "plane"]
 };
 
 var destination4 = {
-    destinationName: "Dublin",
-    distanceKms: 350,
-    transportations: ["plane", "ferry"]
+  destinationName: "Dublin",
+  distanceKms: 350,
+  transportations: ["plane", "ferry"]
 };
 
-var travelDestinations = [destination1, destination2, destination3, destination4];
+var travelDestinations = [
+  destination1,
+  destination2,
+  destination3,
+  destination4
+];
 
 /* 
 DO NOT EDIT ANYTHING ABOVE THIS LINE
 WRITE YOUR CODE BELOW
 */
+var destinationNamesWithin500Kms = travelDestinations
+  .filter(destination => destination.distanceKms < 500)
+  .map(destination => destination.destinationName);
 
+var destinationNameReachableByFerry = travelDestinations
+  .filter(reach => reach.transportations.includes("ferry"))
+  .map(destination => destination.destinationName);
 
-var destinationNamesWithin500Kms = // Complete here
-
-var destinationNameReachableByFerry = // Complete here
-
-var destinationNamesMoreThan300KmsAwayByTrain = // Complete here (PRINT THE RESULT IN THE CONSOLE USING FOREACH)
+travelDestinations.forEach;
 
+var destinationNamesMoreThan300KmsAwayByTrain = travelDestinations
+  .filter(
+    destination =>
+      destination.distanceKms > 300 &&
+      destination.transportations.includes("train")
+  )
+  .map(destinationName => destinationName.destinationName);
```

Comment:
`destinationName.destinationName`... doesn't make sense.  The name of a destination (perhaps 'Paris') is a string and strings don't have a `name` property, in the same way that *your* name doesn't have a name.
Tip: if you find yourself writing phrases which sound nonsensical, stop and check that your variables are well named and that your logic is good.

But, it works, well done!  Now try to make it read better.